packages/client-generator-ts/src/utils/buildGetWasmModule.ts

```diff
@@ -0,0 +1,68 @@
+import { capitalize } from '@prisma/client-common'
+import { ActiveConnectorType } from '@prisma/generator'
+import { match } from 'ts-pattern'
+
+import { RuntimeTarget } from '../runtime-targets'
+import { RuntimeName } from '../TSClient/TSClient'
+
+export type BuildWasmModuleOptions = {
+  component: 'engine' | 'compiler'
+  runtimeName: RuntimeName
+  runtimeBase: string
+  target: RuntimeTarget
+  activeProvider: ActiveConnectorType
+}
+
+export function buildGetWasmModule({
+  component,
+  runtimeName,
+  runtimeBase,
+  target,
+  activeProvider,
+}: BuildWasmModuleOptions) {
+  const capitalizedComponent = capitalize(component)
+
+  const wasmPathBase = `${runtimeBase}/query_compiler_bg.${activeProvider}`
+  const wasmBindingsPath = `${wasmPathBase}.mjs`
+  const wasmModulePath = `${wasmPathBase}.wasm`
+
+  const buildNodeLoader = match(runtimeName)
+    .with('library', () => component === 'engine' && !!process.env.PRISMA_CLIENT_FORCE_WASM)
+    .with('client', () => component === 'compiler')
+    .otherwise(() => false)
+
+  // We're missing an edge bundle for client engine right now
```

Comment:
```suggestion
  // TODO: We're missing an edge bundle for client engine right now
```
Will we need one though? Or can we "rest" on the fact that we have the query compiler + driver adapters soon enough and it's all compatible out of the box? 🤔

Reply:
Yes, we still need one because we can't use the Node.js APIs and filesystem there